mne_bids/path.py

```diff
@@ -1951,8 +1951,8 @@ def _filter_fnames(fnames, *, subject=None, session=None, task=None,
                  if split else r'(|_split-([^_]+))')
     desc_str = (r'_desc-(' + '|'.join(description) + ')'
                 if description else r'(|_desc-([^_]+))')
-    suffix_str = (r'_(' + '|'.join(suffixe) + ')' if suffix
-                  else (r'_(' + '|'.join(ALLOWED_FILENAME_SUFFIX) + ')'))
```

Comment:
I noticed that the function `_filter_fnames()` only returns paths with valid BIDS suffixes even if the two mother functions `find_matching_paths()` and `BIDSPath.match()` have the keyword argument `check=False`. 

In theory, one could add the argument `check=False` to `_filter_fnames(check=False)` and then code 
```Python
    if suffix:
        suffix_str = r'_(' + '|'.join(suffix) + ')'
    elif suffix is None and check:
        # only return files with suffixes that are valid in BIDS
        suffix_str = (r'_(' + '|'.join(ALLOWED_FILENAME_SUFFIX) + ')'))
    elif suffix is None and not check:
        suffix_str = r'_([^_]+)'
```
However, this seems overkill to me since the BIDS conformity is checked afterwards using `_fnames_to_bidspaths(check=check)`.